author: Brian Nelson <pyro@debian.org>

Ensures qmake generates Makefiles that use the -qt4 tools and that use
recursively link against all indirectly-used libraries.

--- qt4-x11-4.3.1.orig/mkspecs/common/linux.conf
+++ qt4-x11-4.3.1/mkspecs/common/linux.conf
@@ -23,8 +23,8 @@
 QMAKE_LIBS_OPENGL_QT  = -lGL
 QMAKE_LIBS_THREAD     = -lpthread
 
-QMAKE_MOC             = $$[QT_INSTALL_BINS]/moc
-QMAKE_UIC             = $$[QT_INSTALL_BINS]/uic
+QMAKE_MOC             = $$[QT_INSTALL_BINS]/moc-qt4
+QMAKE_UIC             = $$[QT_INSTALL_BINS]/uic-qt4
 
 QMAKE_AR              = ar cqs
 QMAKE_RANLIB          =
--- qt4-x11-4.3.1.orig/mkspecs/linux-g++/qmake.conf
+++ qt4-x11-4.3.1/mkspecs/linux-g++/qmake.conf
@@ -4,8 +4,9 @@
 
 MAKEFILE_GENERATOR	= UNIX
 TEMPLATE		= app
-CONFIG			+= qt warn_on release incremental link_prl
+CONFIG			+= qt warn_on release incremental
 QT			+= core gui
+DEFINES		+= QT_SHARED
 QMAKE_INCREMENTAL_STYLE = sublib
 
 include(../common/g++.conf)
